lara-typescript/src/interactive-api-host/attachments-api/handle-get-attachment-url.ts

```diff
@@ -16,35 +18,37 @@ export const handleGetAttachmentUrl =
     return response;
   }
 
-  const answerMeta = options.answerMeta;
-  let { attachmentsFolder, attachments } = answerMeta;
+  // Do not clone the whole answerMeta, as it can be a large object (for example, AP provides the whole Firestore doc).
+  const answerMeta: IAnswerMetadataWithAttachmentsInfo = {
+    attachmentsFolder: clone(options.answerMeta.attachmentsFolder),
+    attachments: clone(options.answerMeta.attachments)
+  };
   try {
     if (operation === "write") {
       if (!attachmentsMgr.isWriteSupported() || !options.writeOptions) {
         response.error = "error getting attachment url: the write operation is not supported by the host environment";
         return response;
       }
-      if (!attachmentsFolder) {
-        attachmentsFolder = answerMeta.attachmentsFolder =
-          await attachmentsMgr.createFolder(options.writeOptions?.interactiveId);
+      if (!answerMeta.attachmentsFolder) {
+        answerMeta.attachmentsFolder = await attachmentsMgr.createFolder(options.writeOptions?.interactiveId);
       }
-      if (!attachments) {
-        attachments = answerMeta.attachments = {};
+      if (!answerMeta.attachments) {
+        answerMeta.attachments = {};
       }
       const urlOptions: ISignedWriteUrlOptions = { ContentType: contentType, expiresIn };
-      const [writeUrl, attachmentInfo] = await attachmentsMgr.getSignedWriteUrl(attachmentsFolder, name, urlOptions);
+      const [writeUrl, attachmentInfo] =
+        await attachmentsMgr.getSignedWriteUrl(answerMeta.attachmentsFolder, name, urlOptions);
       response.url = writeUrl;
       // public path changes with sessionId
-      if (!attachments[name] || (attachmentInfo.publicPath !== attachments[name].publicPath)) {
-        attachments[name] = attachmentInfo;
+      if (!answerMeta.attachments[name] || answerMeta.attachments[name].publicPath !== attachmentInfo.publicPath) {
+        answerMeta.attachments[name] = attachmentInfo;
         options.writeOptions.onAnswerMetaUpdate(answerMeta);
       }
     }
     else if (operation === "read") {
-      if (attachmentsFolder && attachments && attachments[name]) {
+      if (answerMeta.attachmentsFolder && answerMeta.attachments && answerMeta.attachments[name]) {
         // TODO: this won't work for run-with-others where we won't have a readWriteToken
-        const attachmentInfo = { ...attachments[name], folder: attachmentsFolder };
-        response.url = await attachmentsMgr.getSignedReadUrl(attachmentInfo, { expiresIn });
+        response.url = await attachmentsMgr.getSignedReadUrl(answerMeta.attachments[name], { expiresIn });
```

Comment:
I tried to simplify things, but I think this is actually breaking readWriteToken case. Based on the TS types/interfaces, this line:
`const attachmentInfo = { ...attachments[name], folder: attachmentsFolder };`
didn't make too much sense, so I just removed it. But now I'm seeing that `attachmentsFolder` might include read write token that is never part of the `answerMeta.attachments[name]`. I'll try to clarify interfaces so TS compiler can find this bug too.

Reply:
This is fixed here: https://github.com/concord-consortium/lara/pull/754